document torch.testing.assert_allclose

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89559
* #89527
* #89620
* __->__ #89526
* #89525

After our failed attempt to remove `assert_allclose` in #87974, we decided to add it to the documentation after all. Although we drop the expected removal date, the function continues to be deprecated in favor of `assert_close`.

pmeier committed:
document torch.testing.assert_allclose

[ghstack-poisoned]

pmeier committed:
Update on "document torch.testing.assert_allclose"




[ghstack-poisoned]

pmeier committed:
Update on "document torch.testing.assert_allclose"




[ghstack-poisoned]

pmeier commented on torch/testing/_comparison.py:
```diff
@@ -102,7 +105,8 @@ def default_tolerances(
                 f"Expected a torch.Tensor or a torch.dtype, but got {type(input)} instead."
             )
-    rtols, atols = zip(*[_DTYPE_PRECISIONS.get(dtype, (0.0, 0.0)) for dtype in dtypes])
+    dtype_precisions = dtype_precisions or _DTYPE_PRECISIONS
```
Instead of hardcoding `_DTYPE_PRECISIONS` here, this allows us to also use this function for `assert_allclose`, since that uses different default precisions.

mruberry commented on torch/testing/_comparison.py:
```diff
@@ -1531,3 +1535,55 @@ def assert_close(
+    """
+    .. warning::
+
+       :func:`torch.testing.assert_allclose` is deprecated since ``1.12``. Please use
```
"deprecated since PyTorch 1.12 and will be removed in a future release."

pmeier replied:
Oh, you have some optimism left :smile:

mruberry commented on torch/testing/_comparison.py:
```diff
@@ -1531,3 +1535,55 @@ def assert_close(
+       `here <https://github.com/pytorch/pytorch/issues/61844>`_.
+    """
+    warnings.warn(
+        "`torch.testing.assert_allclose()` is deprecated since 1.12. "
```
Same change here, please

mruberry reviewed (approved):
Stamped, see one inline suggestion

pmeier committed:
Update on "document torch.testing.assert_allclose"


After our failed attempt to remove `assert_allclose` in #87974, we decided to add it to the documentation after all. Although we drop the expected removal date, the function continues to be deprecated in favor of `assert_close`.


[ghstack-poisoned]
